## Implement linear SVM method using scikit library

In [7]:
# importing all the required libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn import metrics
from sklearn.metrics import classification_report
from sklearn.svm import SVC

## Read the glass dataset

In [8]:
# Read dataset
data = pd.read_csv('/content/glass.csv')
X = data.drop('Type', axis=1)
y = data['Type'].values

## Split the data into Training and Test set

In [9]:
# Use train_test_split to split data into the Training set and Test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=0)

## Fit the Linear SVM model

In [10]:
# Model Training set into linear SVM
fitSVC = SVC(kernel='linear')
fitSVC.fit(X_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='linear',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)

## Evaluate the model on testing part using score

In [11]:
# Predict the Test set results
y_pred = fitSVC.predict(X_test)

# Calculate Accuracy of SVM Model
print('Accuracy of linear SVM model is: ', metrics.accuracy_score(y_test, y_pred))

Accuracy of linear SVM model is:  0.5555555555555556


## Use the model to predict the classes of the validation set & mean accuracy.

In [12]:
# Print the classification report
print(classification_report(y_test, y_pred, zero_division=1))

              precision    recall  f1-score   support

           1       0.43      0.86      0.57        14
           2       0.60      0.38      0.46        24
           3       1.00      0.00      0.00         5
           5       0.67      1.00      0.80         2
           6       0.00      0.00      0.00         2
           7       1.00      1.00      1.00         7

    accuracy                           0.56        54
   macro avg       0.62      0.54      0.47        54
weighted avg       0.62      0.56      0.51        54



## From the Classification Report, Accuracy is around ~56% for linear SVM Model.
## Comparing both the results, it is clear that linear SVM is getting us the better accuracy.
### **Justification**: The possible reason could be Naive Bayes treats 'features' as independent, whereas SVM looks at the interactions between the features to a certain degree like linear, guasian etc., which helps in achieving better accuracy.